### PR TITLE
the deep fryer reacts to cold

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

dunking a colder than 0 celsius person in the deep fryer causes additional damage
dunking a VERY cold person causes fun stuff
putting a freeze cube in the deep fryer causes !FUN! stuff
v

## Why It's Good For The Game

https://www.youtube.com/watch?v=Yu7vM7jyomg

## Changelog

:cl:
expansion: dunking cold people in the deep fryer causes fun stuff
expansion: dunking the Freeze Cube artifact from lavaland causes !FUN! stuff
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 1]
```diff
@@ -130,16 +127,43 @@ GLOBAL_LIST_INIT(oilfry_blacklisted_items, typecacheof(list(
...
+		dunking_target.Paralyze(60)
 		user.changeNext_move(CLICK_CD_MELEE)
 	return ..()
 
+/obj/machinery/deepfryer/proc/fry(obj/item/frying_item, mob/user)
+	to_chat(user, span_notice("You put [frying_item] into [src]."))
+	if(istype(frying_item, /obj/item/freeze_cube))
+		log_bomber(user, "put a freeze cube in a", src)
+		visible_message(span_userdanger("[src] starts glowing... Oh no..."))
+		playsound(src, 'sound/effects/pray_chaplain.ogg, 100)
```
Comment (tralezab) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:156]: you did not even compile your code

```suggestion
		playsound(src, 'sound/effects/pray_chaplain.ogg', 100)
```

**Review by tralezab — CHANGES_REQUESTED** [review]
i will give a full review when you give a smidgen of effort testing ur code 😤

**Comment by Fikou** [comment]
> i will give a full review when you give a smidgen of effort testing ur code 😤

wdym therse a screenshot in the pr of the code

**Comment by carshalash** [comment]
Fikou no.

**Comment by MrMelbert** [comment]
is it possible to make ~~turkey~~ chicken meat naturally cold and/or naturally react in the fryer

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 2]
```diff
@@ -130,16 +127,43 @@ GLOBAL_LIST_INIT(oilfry_blacklisted_items, typecacheof(list(
...
+		add_filter("entropic_ray", 10, list("type" = "rays", "size" = 35, "color" = COLOR_VIVID_YELLOW))
+		addtimer(CALLBACK(src, .proc/blow_up), 5 SECONDS)
+	frying = new /obj/item/food/deepfryholder(src, frying_item)
+	icon_state = "fryer_on"
+	fry_loop.start()
+
+/obj/machinery/deepfryer/proc/blow_up()
+	visible_message(span_userdanger("[src] blows up from the entropic reaction!"))
+	explosion(src, devastation_range = 1, heavy_impact_range = 3, light_impact_range = 5, flame_range = 7)
+	qdel(src)
```
Comment (LemonInTheDark) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:166]: why explode and qdel?

Comment (Fikou) [reply code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:166]: some other things that did explosions exploded and then qdeld so i just copied that

File: code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm [hunk 3]
```diff
@@ -130,16 +127,43 @@ GLOBAL_LIST_INIT(oilfry_blacklisted_items, typecacheof(list(
...
+		var/mob/living/carbon/dunking_target = user.pulling
+		log_combat(user, dunking_target, "dunked", null, "into [src]")
+		user.visible_message(span_danger("[user] dunks [dunking_target]'s face in [src]!"))
+		reagents.expose(dunking_target, TOUCH)
+		var/permeability = 1 - dunking_target.get_permeability_protection(list(HEAD))
+		var/target_temp = dunking_target.bodytemperature
+		var/cold_multiplier = 1
+		if(target_temp < TCMB + 10) // a tiny bit of leeway
+			dunking_target.visible_message(span_userdanger("[dunking_target] explodes from the entropic difference! Holy fuck!"))
+			dunking_target.gib()
```
Comment (Ghommie) [inline code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm:139]: Shouldn't the head be the only exploding bit here considering it's also only part of the target that's being dunked in the fryer?
On a sidenote, achievement?

**Comment by ATH1909** [comment]
what about the ice reagent?

**Pull request merged**